Orange/widgets/data/owselectrows.py

```diff
@@ -530,8 +530,11 @@ def commit(self):
                 matching_output = remover(matching_output)
                 non_matching_output = remover(non_matching_output)
 
-        self.Outputs.matching_data.send(matching_output)
-        self.Outputs.unmatched_data.send(non_matching_output)
+        def output_filter(output):
+            return output if output is not None and len(output) else None
```

Comment:
I here agree with @astaric's usual comment about my code. If-else expressions in Python are mostly unreadable. I'd prefer

```
if output is not None and len(output):
    return output
```

You don't even need `return None` here, but you may add it to make it explicit.

Reply:
Oh, besides, I don't like the function at all. I'd prefer having

```
if not len(matching_output):
    matching_output = None
```

and the same for `non_matching_output`.

The same amount of code, but way more explicit.